packages/@react-aria/loading/src/useLoadOnScroll.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {RefObject, useCallback, useMemo, useRef} from 'react';
+import {useLayoutEffect} from '@react-aria/utils';
+
+export interface LoadOnScrollProps {
+  /** Whether data is currently being loaded. */
+  isLoading?: boolean,
+  /** Handler that is called when more items should be loaded, e.g. while scrolling near the bottom.  */
+  onLoadMore?: () => void,
+  // TODO: decide on default here
+  /**
+   * The amount of offset (in pixels) from the bottom of your scrollable region that should trigger load more.
+   * @default 25
+   */
+  scrollOffset?: number
```

Comment:
Open question, I could make this default to 0 I guess? Maybe a percentage of the scroll body height?

Reply:
In virtualizer this defaulted to 2 page heights from the bottom I believe. It does seem like incorporating the the height would be useful as a trigger for loading more. 25px is quite small if loading is slow.

Reply:
should it just be based off using the ref and querying the client height then? Just worried about the case where the ref isn't defined on first render and thus you get a default that doesn't exist if that makes sense

Reply:
updated to be a percentage value based of the clientHeight of your ref